File: LayoutTests/http/tests/security/contentSecurityPolicy/eval-allowed-in-report-only-mode.php
<?php
header("Content-Security-Policy-Report-Only: script-src 'self'");
?>
<!DOCTYPE html>
<html>
<body>
<script>
if (window.testRunner)
    testRunner.dumpAsText();

eval("alert('PASS: eval() executed as expected.');");
</script>
</body>
</html>

File: LayoutTests/http/tests/security/contentSecurityPolicy/eval-allowed-in-report-only-mode.py
#!/usr/bin/env python3

import sys

sys.stdout.write(
    'Content-Security-Policy-Report-Only: script-src \'self\'\r\n'
    'Content-Type: text/html\r\n\r\n'
    '<!DOCTYPE html>\n'
    '<html>\n'
    '<body>\n'
    '<script>\n'
    'if (window.testRunner)\n'
    '    testRunner.dumpAsText();\n'
    '\n'
    'eval("alert(\'PASS: eval() executed as expected.\');");\n'
    '</script>\n'
    '</body>\n'
    '</html>\n'
)